### PR TITLE
Susan/general bug fixes

### PR DESCRIPTION
**Pull request by sadelokiki**
Form validation bulk upload
Format error message for Bulk import
Add IDSeq mail access link
Update search UI fonts and layout
Add formatted error responses for unhandled exceptions from by db upgrade

### CODE REVIEW

File: app/assets/javascripts/components/bulkUploadImport.jsx [hunk 1]
```diff
@@ -192,28 +199,26 @@ class BulkUploadImport extends React.Component {
...
       that.initializeSelectTag()
       that.initializeSelectAll()
-    }).catch(function (error) {
+    })
+    .catch((error) => {
      that.setState({
       submitting: false,
       invalid: true,
-      errorMessage: JSON.stringify(error.response)
+      errorMessage: error.response.data.status || 'Something went wrong'
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/bulkUploadImport.jsx]: what does the status tell us from the server side?

File: app/assets/javascripts/components/bulkUploadImport.jsx [hunk 2]
```diff
@@ -272,7 +300,8 @@ class BulkUploadImport extends React.Component {
     const samples = this.state.samples
     samples[e.target.id].project_id = this.state.allProjects[e.target.selectedIndex].id
     this.setState({
-      samples: samples
+      samples: samples,
+      project: e.target.value.trim()
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/bulkUploadImport.jsx]: what is this line for? why change project when you just change project for one sample.

File: app/assets/javascripts/components/bulkUploadImport.jsx [hunk 3]
```diff
@@ -327,7 +364,7 @@ class BulkUploadImport extends React.Component {
                     <label htmlFor={i}> {sample.name}</label>
                   </p>
                   <div className="col s4">
-                    <select className="" id={i} onChange={ this.handleProjectChangeForSample } value={sample.project_id}>
+                    <select className="" id={i} onChange={ this.handleProjectChangeForSample } value={this.state.project}>
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/bulkUploadImport.jsx]: why do you set the value to this.state.project? individual samples can map to different project id.

File: app/assets/javascripts/components/bulkUploadImport.jsx [hunk 4]
```diff
@@ -424,7 +463,8 @@ class BulkUploadImport extends React.Component {
   render() {
     return (
       <div>
-        { this.state.imported ? this.renderBulkUploadSubmitForm(): this.renderBulkUploadImportForm() }
+        { this.state.imported ? this.renderBulkUploadSubmitForm() : null }
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/bulkUploadImport.jsx]: what's the reason of separating them into different lines?

File: app/assets/javascripts/components/bulkUploadImport.jsx (hunk 1 above)
Comment (sadelokiki) [reply app/assets/javascripts/components/bulkUploadImport.jsx]: it returns the specific error object returned from the server - 'No samples imported under bulk_path'.  @yunfangjuan

File: app/assets/javascripts/components/bulkUploadImport.jsx (hunk 3 above)
Comment (sadelokiki) [reply app/assets/javascripts/components/bulkUploadImport.jsx]: Just noticed that all sample project ids are being updated on project change, so no need to change then individually, i would revert it, thanks.

File: app/assets/javascripts/components/bulkUploadImport.jsx (hunk 4 above)
Comment (sadelokiki) [reply app/assets/javascripts/components/bulkUploadImport.jsx]: having them on both sides of the ternary operator was throwing a DOM error poiting to hiding and not completely removing the component(after rendering), so I decided to either show or return null for both components as opposed to hiding. @yunfangjuan

**Pull request merged**